### PR TITLE
[Build/Testing/CI]install.sh and docker release

### PR DESCRIPTION
**Pull request by ZhiHanZ**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

Summary about this PR
Instroduce install.sh for datafuse 
will install fuse-query and fuse-store in ~/.datafuse/bin by default and supports to detect on different platforms

### How to use install.sh
```bash
Usage: $this [-b] bindir [-d] [tag]
  -b sets bindir or installation directory, Defaults to {HOME}/.datafuse/bin
  -d turns on debug logging
   [tag] is a tag from
   https://github.com/datafuselabs/datafuse/releases
   If tag is missing, then the latest will be used.
```
for example
```bash
./install.sh v0.4.33-nightly will download given tag release to ~/.datafuse/bin directory if operating system and platform supported
```
if there is no version given, it will download the latest release

### Some issues and example build image
I have tested manually on M1 chip, works fine for arm64 and amd64 architecture others(armv7 and armv6) have some errors, could be solve by using some physical machine
```bash
#12 0.632 /usr/bin/head: error reading '/proc/self/exe': Bad file descriptor
#12 0.632 /usr/bin/head: failed to close '/proc/self/exe': Bad file descriptor
```
This one is build by release docker
https://hub.docker.com/layers/159066121/zhihanz/datafuse-release/latest/images/sha256-37af7d525ef4261d2ca70115ff38e1f6085ee0ef6b15a312e06a89bd70c49873?context=repo
## Changelog

- New Feature
- Build/Testing/CI

## Related Issues
Partial resolve https://github.com/datafuselabs/datafuse/issues/1114

## Test Plan
Unit Tests

Stateless Tests

### CODE REVIEW

File: scripts/install.sh [hunk 1]
```diff
@@ -0,0 +1,599 @@
+#!/bin/bash
```
Comment (BohuTANG) [inline scripts/install.sh:1]: Cool.
I prefer to move this shell to a new director 'installer' and rename to 'install-datafuse.sh'.
install.sh will be the default script for installing the tool of datafuse cli, then we can install/upgrade the datafuse binaries via the cli

File: docker/release/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM debian:buster as installer
+COPY ./scripts/install.sh /install.sh
```
Comment (BohuTANG) [inline docker/release/Dockerfile:2]: installer/install-datafuse.sh？

Comment (ZhiHanZ) [reply docker/release/Dockerfile]: release image updated

**Review by BohuTANG — APPROVED** [review]
LGTM

**Comment by BohuTANG** [comment]
Oh, we should move the installer from the root to scripts next time.

**Comment by BohuTANG** [comment]
OS:
```uname -a
Linux thinkpad 5.10.0-1033-oem #34-Ubuntu SMP Fri Jun 18 14:57:41 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux```

Error:
```
 ./installer/install-datafuse.sh 
./installer/install-datafuse.sh err need 'jq' (command not found)
```

**Comment by BohuTANG** [comment]
Can we give a download progress bar here, it's also useful for download datafuse-cli:

```
./installer/install-datafuse.sh 
./installer/install-datafuse.sh info 👏👏👏 Welcome to use datafuse!
sysctl: invalid option -- 'i'
Running on native Intel
./installer/install-datafuse.sh info 😊 Your Architecture x86_64-unknown-linux-gnu is supported
https://github.com/datafuselabs/datafuse/releases/download/v0.4.56-nightly/datafuse--x86_64-unknown-linux-gnu.tar.gz
```

**Pull request merged**